observ/store.py

```diff
@@ -21,7 +22,7 @@ def inner(self, *args, **kwargs):
         readonly_state = self.state
         self.state = self._present
         try:
-            current = to_raw(self.state)
+            current = deepcopy(to_raw(self.state))
             fn(self, *args, **kwargs)
             ops, reverse_ops = patchdiff.diff(current, self.state)
```

Comment:
you might want to check/test how this behaves in the problem case described in the xfailing test

Reply:
yeah, I'm leaning towards removing the to_raw from the write traps and just restore the to_raw function. 
Because within our traps it doesn't really matter apparently, and the to_raw function will then ensure a non-proxied object if needed (such as in the store). And then I'll remove the deepcopy again from the store.